b_cfn_elasticsearch_cloner/resource.py

```diff
@@ -0,0 +1,211 @@
+
+        initial_cloner_function = SingletonFunction(
+            scope=self,
+            id="InitialClonerFunction",
+            uuid="e01116a4-f939-43f2-8f5b-cc9f862c9e01",
+            lambda_purpose="InitialClonerSingletonLambda",
+            code=Code.from_asset(initial_cloner_root),
+            handler="index.handler",
+            runtime=Runtime.PYTHON_3_8,
+            layers=[ElasticsearchLayer],
+            log_retention=RetentionDays.ONE_MONTH,
+            memory_size=128,
+            role=Role(
+                scope=self,
+                id="InitialClonerFunctionRole",
+                assumed_by=ServicePrincipal("lambda.amazonaws.com"),
+                inline_policies={
+                    "LogsPolicy": PolicyDocument(
+                        statements=[
+                            PolicyStatement(
+                                actions=[
+                                    "logs:CreateLogGroup",
+                                    "logs:CreateLogStream",
+                                    "logs:PutLogEvents",
+                                    "logs:DescribeLogStreams",
+                                ],
+                                resources=["arn:aws:logs:*:*:*"],
+                                effect=Effect.ALLOW,
+                            )
+                        ]
+                    ),
+                    "ElasticsearchPolicy": PolicyDocument(
+                        statements=[
+                            PolicyStatement(
+                                actions=[
+                                    "es:ESHttpDelete",
+                                    "es:ESHttpGet",
+                                    "es:ESHttpHead",
+                                    "es:ESHttpPatch",
+                                    "es:ESHttpPost",
+                                    "es:ESHttpPut",
+                                ],
+                                resources=["*"],
+                                effect=Effect.ALLOW,
+                            )
+                        ]
+                    ),
+                    "DynamodbPolicy": PolicyDocument(
+                        statements=[
+                            PolicyStatement(
+                                actions=["dynamodb:*"],
+                                resources=["*"],
+                                effect=Effect.ALLOW,
+                            )
+                        ]
+                    ),
+                },
+                description="Role for DynamoDB Initial Cloner Function",
+            ),
+        )
+
+        if kms_key:
+            initial_cloner_function.add_to_role_policy(
+                PolicyStatement(
+                    actions=["kms:Decrypt"],
+                    resources=[kms_key.key_arn],
+                    effect=Effect.ALLOW,
+                ),
+            )
+
+        initial_cloner = CustomResource(
+            scope=self,
+            id="InitialCloner",
```

Comment:
Do not hardcode ids. Two same resources would result in a CDK error. Instead of taking an "id" in the constructor, take a prefix and add that prefix to all of your resource names and ids.